admin/services/pin-generation.service.js

```diff
@@ -80,7 +61,7 @@ pinGenerationService.generateSchoolPassword = school => {
     wordsArray[pinGenerationService.generateCryptoRandomNumber(0, wordsArray.length - 1)]
   const numberCombination = randomGenerator.getRandom(2, chars)
   const newPin = `${firstRandomWord}${numberCombination}${secondRandomWord}`
-  const newExpiry = pinExpiryTime(school.timezone)
+  const newExpiry = pinTimestampService.generatePinTimestamp(config.OverridePinExpiry, endOfDay, fourPmToday, school.timezone)
```

Comment:
There seem to be too many arguments here.  Surely, the first 3 aren't actually needed?

It can get the config itself.
endfOfDay and fourPmToday are derived from time, although passing them in as args may assist in testing.  Is that why?

Reply:
as discussed, i am resolving this.